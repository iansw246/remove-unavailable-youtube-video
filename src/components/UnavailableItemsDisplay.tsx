import { Typography } from "@mui/material";
import { Playlist, PlaylistItem } from "../utils/requestHelpers";
import PlaylistItemList from "./PlaylistItemList";

export interface Props {
    unavailableItems: PlaylistItem[];
    playlist?: Playlist;
}

export default function UnavailableItemsDisplay({ unavailableItems, playlist }: Props) {
    return (
        <>
            <Typography>Found {unavailableItems?.length ?? 0} unavailable videos {playlist ? `in playlist ${playlist.snippet?.title}` : ""}</Typography>
            { unavailableItems && <PlaylistItemList items={unavailableItems} /> }
        </>
    );
}