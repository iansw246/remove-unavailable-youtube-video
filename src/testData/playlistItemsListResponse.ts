// Response from https://content-youtube.googleapis.com/youtube/v3/playlistItems?playlistId=OLAK5uy_lVHKpWYVeLJEP2eqMjfCk0uYE54qRXo10&maxResults=50&part=contentDetails%2Cid%2Csnippet%2Cstatus&key=AIzaSyAa8yy0GdcGPHdtD083HiGGx_S0vMPScDM

const response = {
  "kind": "youtube#playlistItemListResponse",
  "etag": "OlXS_VhHXCUF7dva8dYTY4HwCao",
  "items": [
    {
      "kind": "youtube#playlistItem",
      "etag": "Bk82WivhrcU7VXe9qv5pvNIKm-w",
      "id": "T0xBSzV1eV9sVkhLcFdZVmVMSkVQMmVxTWpmQ2swdVlFNTRxUlhvMTAuMTNGMkFGMUE4OTk4MkNGOQ",
      "snippet": {
        "publishedAt": "2023-09-29T03:18:04Z",
        "channelId": "UCBR8-60-B28hp2BmDPdntcQ",
        "title": "Boat Buds",
        "description": "Provided to YouTube by CDBaby\n\nBoat Buds · Mark Sparling\n\nBoat Buds: A Short Hike EP\n\n℗ 2021 Mark Sparling\n\nReleased on: 2021-01-18\n\nAuto-generated by YouTube.",
        "thumbnails": {
          "default": {
            "url": "https://i.ytimg.com/vi/Dc30Pk_Ib1w/default.jpg",
            "width": 120,
            "height": 90
          },
          "medium": {
            "url": "https://i.ytimg.com/vi/Dc30Pk_Ib1w/mqdefault.jpg",
            "width": 320,
            "height": 180
          },
          "high": {
            "url": "https://i.ytimg.com/vi/Dc30Pk_Ib1w/hqdefault.jpg",
            "width": 480,
            "height": 360
          },
          "standard": {
            "url": "https://i.ytimg.com/vi/Dc30Pk_Ib1w/sddefault.jpg",
            "width": 640,
            "height": 480
          },
          "maxres": {
            "url": "https://i.ytimg.com/vi/Dc30Pk_Ib1w/maxresdefault.jpg",
            "width": 1280,
            "height": 720
          }
        },
        "channelTitle": "YouTube",
        "playlistId": "OLAK5uy_lVHKpWYVeLJEP2eqMjfCk0uYE54qRXo10",
        "position": 0,
        "resourceId": {
          "kind": "youtube#video",
          "videoId": "Dc30Pk_Ib1w"
        },
        "videoOwnerChannelTitle": "Mark Sparling - Topic",
        "videoOwnerChannelId": "UCPw7S3ZvjuPnc2-KCIxZYbw"
      },
      "contentDetails": {
        "videoId": "Dc30Pk_Ib1w",
        "videoPublishedAt": "2021-01-17T10:06:03Z"
      },
      "status": {
        "privacyStatus": "public"
      }
    },
    {
      "kind": "youtube#playlistItem",
      "etag": "FjY6T7mux9Qjmj06xifbrYFeT_4",
      "id": "T0xBSzV1eV9sVkhLcFdZVmVMSkVQMmVxTWpmQ2swdVlFNTRxUlhvMTAuMkI3Q0VEMkNDNDBGQkJBMQ",
      "snippet": {
        "publishedAt": "2023-09-29T03:18:04Z",
        "channelId": "UCBR8-60-B28hp2BmDPdntcQ",
        "title": "Beach Buds (Flight Mix)",
        "description": "Provided to YouTube by CDBaby\n\nBeach Buds (Flight Mix) · Mark Sparling\n\nBoat Buds: A Short Hike EP\n\n℗ 2021 Mark Sparling\n\nReleased on: 2021-01-18\n\nAuto-generated by YouTube.",
        "thumbnails": {
          "default": {
            "url": "https://i.ytimg.com/vi/c5CxLUwJATw/default.jpg",
            "width": 120,
            "height": 90
          },
          "medium": {
            "url": "https://i.ytimg.com/vi/c5CxLUwJATw/mqdefault.jpg",
            "width": 320,
            "height": 180
          },
          "high": {
            "url": "https://i.ytimg.com/vi/c5CxLUwJATw/hqdefault.jpg",
            "width": 480,
            "height": 360
          },
          "standard": {
            "url": "https://i.ytimg.com/vi/c5CxLUwJATw/sddefault.jpg",
            "width": 640,
            "height": 480
          },
          "maxres": {
            "url": "https://i.ytimg.com/vi/c5CxLUwJATw/maxresdefault.jpg",
            "width": 1280,
            "height": 720
          }
        },
        "channelTitle": "YouTube",
        "playlistId": "OLAK5uy_lVHKpWYVeLJEP2eqMjfCk0uYE54qRXo10",
        "position": 1,
        "resourceId": {
          "kind": "youtube#video",
          "videoId": "c5CxLUwJATw"
        },
        "videoOwnerChannelTitle": "Mark Sparling - Topic",
        "videoOwnerChannelId": "UCPw7S3ZvjuPnc2-KCIxZYbw"
      },
      "contentDetails": {
        "videoId": "c5CxLUwJATw",
        "videoPublishedAt": "2021-01-17T10:04:25Z"
      },
      "status": {
        "privacyStatus": "public"
      }
    },
    {
      "kind": "youtube#playlistItem",
      "etag": "EKUNTRtzv2kV9zqfbEHf3MhAR_g",
      "id": "T0xBSzV1eV9sVkhLcFdZVmVMSkVQMmVxTWpmQ2swdVlFNTRxUlhvMTAuRUZCMzZCRUJEN0ZFMDlGOQ",
      "snippet": {
        "publishedAt": "2023-09-29T03:18:04Z",
        "channelId": "UCBR8-60-B28hp2BmDPdntcQ",
        "title": "Somewhere in the Woods (Flight Mix)",
        "description": "Provided to YouTube by CDBaby\n\nSomewhere in the Woods (Flight Mix) · Mark Sparling\n\nBoat Buds: A Short Hike EP\n\n℗ 2021 Mark Sparling\n\nReleased on: 2021-01-18\n\nAuto-generated by YouTube.",
        "thumbnails": {
          "default": {
            "url": "https://i.ytimg.com/vi/Ezl0eDxazeY/default.jpg",
            "width": 120,
            "height": 90
          },
          "medium": {
            "url": "https://i.ytimg.com/vi/Ezl0eDxazeY/mqdefault.jpg",
            "width": 320,
            "height": 180
          },
          "high": {
            "url": "https://i.ytimg.com/vi/Ezl0eDxazeY/hqdefault.jpg",
            "width": 480,
            "height": 360
          },
          "standard": {
            "url": "https://i.ytimg.com/vi/Ezl0eDxazeY/sddefault.jpg",
            "width": 640,
            "height": 480
          },
          "maxres": {
            "url": "https://i.ytimg.com/vi/Ezl0eDxazeY/maxresdefault.jpg",
            "width": 1280,
            "height": 720
          }
        },
        "channelTitle": "YouTube",
        "playlistId": "OLAK5uy_lVHKpWYVeLJEP2eqMjfCk0uYE54qRXo10",
        "position": 2,
        "resourceId": {
          "kind": "youtube#video",
          "videoId": "Ezl0eDxazeY"
        },
        "videoOwnerChannelTitle": "Mark Sparling - Topic",
        "videoOwnerChannelId": "UCPw7S3ZvjuPnc2-KCIxZYbw"
      },
      "contentDetails": {
        "videoId": "Ezl0eDxazeY",
        "videoPublishedAt": "2021-01-17T10:04:25Z"
      },
      "status": {
        "privacyStatus": "public"
      }
    },
    {
      "kind": "youtube#playlistItem",
      "etag": "N3cI8yb_wTYS8uIGt29oCUFQ1Sg",
      "id": "T0xBSzV1eV9sVkhLcFdZVmVMSkVQMmVxTWpmQ2swdVlFNTRxUlhvMTAuNDNGODBFQzNDMUU2MURERA",
      "snippet": {
        "publishedAt": "2023-09-29T03:18:04Z",
        "channelId": "UCBR8-60-B28hp2BmDPdntcQ",
        "title": "See You at the Top (Flight Mix)",
        "description": "Provided to YouTube by CDBaby\n\nSee You at the Top (Flight Mix) · Mark Sparling\n\nBoat Buds: A Short Hike EP\n\n℗ 2021 Mark Sparling\n\nReleased on: 2021-01-18\n\nAuto-generated by YouTube.",
        "thumbnails": {
          "default": {
            "url": "https://i.ytimg.com/vi/BJNkEeZezSw/default.jpg",
            "width": 120,
            "height": 90
          },
          "medium": {
            "url": "https://i.ytimg.com/vi/BJNkEeZezSw/mqdefault.jpg",
            "width": 320,
            "height": 180
          },
          "high": {
            "url": "https://i.ytimg.com/vi/BJNkEeZezSw/hqdefault.jpg",
            "width": 480,
            "height": 360
          },
          "standard": {
            "url": "https://i.ytimg.com/vi/BJNkEeZezSw/sddefault.jpg",
            "width": 640,
            "height": 480
          },
          "maxres": {
            "url": "https://i.ytimg.com/vi/BJNkEeZezSw/maxresdefault.jpg",
            "width": 1280,
            "height": 720
          }
        },
        "channelTitle": "YouTube",
        "playlistId": "OLAK5uy_lVHKpWYVeLJEP2eqMjfCk0uYE54qRXo10",
        "position": 3,
        "resourceId": {
          "kind": "youtube#video",
          "videoId": "BJNkEeZezSw"
        },
        "videoOwnerChannelTitle": "Mark Sparling - Topic",
        "videoOwnerChannelId": "UCPw7S3ZvjuPnc2-KCIxZYbw"
      },
      "contentDetails": {
        "videoId": "BJNkEeZezSw",
        "videoPublishedAt": "2021-01-17T10:04:25Z"
      },
      "status": {
        "privacyStatus": "public"
      }
    },
    {
      "kind": "youtube#playlistItem",
      "etag": "AbKB4j6nV_vYoEmtMLMAvTFbRpA",
      "id": "T0xBSzV1eV9sVkhLcFdZVmVMSkVQMmVxTWpmQ2swdVlFNTRxUlhvMTAuODhCNzVENEE5RTNBQjlFOA",
      "snippet": {
        "publishedAt": "2023-09-29T03:18:04Z",
        "channelId": "UCBR8-60-B28hp2BmDPdntcQ",
        "title": "Boat Buds (Alternate Mix)",
        "description": "Provided to YouTube by CDBaby\n\nBoat Buds (Alternate Mix) · Mark Sparling\n\nBoat Buds: A Short Hike EP\n\n℗ 2021 Mark Sparling\n\nReleased on: 2021-01-18\n\nAuto-generated by YouTube.",
        "thumbnails": {
          "default": {
            "url": "https://i.ytimg.com/vi/TnsmKZmw_QY/default.jpg",
            "width": 120,
            "height": 90
          },
          "medium": {
            "url": "https://i.ytimg.com/vi/TnsmKZmw_QY/mqdefault.jpg",
            "width": 320,
            "height": 180
          },
          "high": {
            "url": "https://i.ytimg.com/vi/TnsmKZmw_QY/hqdefault.jpg",
            "width": 480,
            "height": 360
          },
          "standard": {
            "url": "https://i.ytimg.com/vi/TnsmKZmw_QY/sddefault.jpg",
            "width": 640,
            "height": 480
          },
          "maxres": {
            "url": "https://i.ytimg.com/vi/TnsmKZmw_QY/maxresdefault.jpg",
            "width": 1280,
            "height": 720
          }
        },
        "channelTitle": "YouTube",
        "playlistId": "OLAK5uy_lVHKpWYVeLJEP2eqMjfCk0uYE54qRXo10",
        "position": 4,
        "resourceId": {
          "kind": "youtube#video",
          "videoId": "TnsmKZmw_QY"
        },
        "videoOwnerChannelTitle": "Mark Sparling - Topic",
        "videoOwnerChannelId": "UCPw7S3ZvjuPnc2-KCIxZYbw"
      },
      "contentDetails": {
        "videoId": "TnsmKZmw_QY",
        "videoPublishedAt": "2021-01-17T10:04:24Z"
      },
      "status": {
        "privacyStatus": "public"
      }
    }
  ],
  "pageInfo": {
    "totalResults": 5,
    "resultsPerPage": 50
  }
};

export default response;