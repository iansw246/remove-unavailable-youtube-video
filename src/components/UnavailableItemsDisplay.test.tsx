import { render } from "@testing-library/react";
import playlistItemsListResponse from "../testData/playlistItemsListResponse";
import playlistListResponse from "../testData/playlistsListResponse";
import UnavailableItemsDisplay from "./UnavailableItemsDisplay";

test("Renders correctly", () => {
  const { asFragment } = render(
    <UnavailableItemsDisplay
      unavailableItems={playlistItemsListResponse.items}
      playlist={playlistListResponse.items[0]}
    />,
  );

  expect(asFragment()).toMatchSnapshot();
});

test("Renders playlist item names", () => {
  const { getByText } = render(
    <UnavailableItemsDisplay
      unavailableItems={playlistItemsListResponse.items}
      playlist={playlistListResponse.items[0]}
    />,
  );

  for (const item of playlistItemsListResponse.items) {
    expect(getByText(item.snippet.title)).toBeInTheDocument();
  }
});
