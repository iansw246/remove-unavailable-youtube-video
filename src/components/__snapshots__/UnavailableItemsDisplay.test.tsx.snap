// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    Found 5 unavailable videos in playlist Delete Test
  </p>
  <div>
    <div
      class="MuiStack-root css-1k2zuzg-MuiStack-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 css-1j8pgxe-MuiPaper-root"
      >
        <div
          class="MuiBox-root css-p58oka"
        >
          <img
            alt="Boat Buds thumbnail"
            class="css-wnbi1d"
            loading="lazy"
            src="https://i.ytimg.com/vi/Dc30Pk_Ib1w/default.jpg"
          />
          <div
            class="MuiBox-root css-gqssyb"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
                href="https://www.youtube.com/watch?v=Dc30Pk_Ib1w&list=OLAK5uy_lVHKpWYVeLJEP2eqMjfCk0uYE54qRXo10&position=0"
                rel="noreferrer"
                target="_blank"
              >
                Boat Buds
              </a>
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
                href="https://www.youtube.com/channel/UCPw7S3ZvjuPnc2-KCIxZYbw"
                rel="noreferrer"
                target="_blank"
              >
                Mark Sparling - Topic
              </a>
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Published on 9/28/2023, 8:18:04 PM
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 css-1j8pgxe-MuiPaper-root"
      >
        <div
          class="MuiBox-root css-p58oka"
        >
          <img
            alt="Beach Buds (Flight Mix) thumbnail"
            class="css-wnbi1d"
            loading="lazy"
            src="https://i.ytimg.com/vi/c5CxLUwJATw/default.jpg"
          />
          <div
            class="MuiBox-root css-gqssyb"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
                href="https://www.youtube.com/watch?v=c5CxLUwJATw&list=OLAK5uy_lVHKpWYVeLJEP2eqMjfCk0uYE54qRXo10&position=1"
                rel="noreferrer"
                target="_blank"
              >
                Beach Buds (Flight Mix)
              </a>
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
                href="https://www.youtube.com/channel/UCPw7S3ZvjuPnc2-KCIxZYbw"
                rel="noreferrer"
                target="_blank"
              >
                Mark Sparling - Topic
              </a>
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Published on 9/28/2023, 8:18:04 PM
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 css-1j8pgxe-MuiPaper-root"
      >
        <div
          class="MuiBox-root css-p58oka"
        >
          <img
            alt="Somewhere in the Woods (Flight Mix) thumbnail"
            class="css-wnbi1d"
            loading="lazy"
            src="https://i.ytimg.com/vi/Ezl0eDxazeY/default.jpg"
          />
          <div
            class="MuiBox-root css-gqssyb"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
                href="https://www.youtube.com/watch?v=Ezl0eDxazeY&list=OLAK5uy_lVHKpWYVeLJEP2eqMjfCk0uYE54qRXo10&position=2"
                rel="noreferrer"
                target="_blank"
              >
                Somewhere in the Woods (Flight Mix)
              </a>
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
                href="https://www.youtube.com/channel/UCPw7S3ZvjuPnc2-KCIxZYbw"
                rel="noreferrer"
                target="_blank"
              >
                Mark Sparling - Topic
              </a>
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Published on 9/28/2023, 8:18:04 PM
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 css-1j8pgxe-MuiPaper-root"
      >
        <div
          class="MuiBox-root css-p58oka"
        >
          <img
            alt="See You at the Top (Flight Mix) thumbnail"
            class="css-wnbi1d"
            loading="lazy"
            src="https://i.ytimg.com/vi/BJNkEeZezSw/default.jpg"
          />
          <div
            class="MuiBox-root css-gqssyb"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
                href="https://www.youtube.com/watch?v=BJNkEeZezSw&list=OLAK5uy_lVHKpWYVeLJEP2eqMjfCk0uYE54qRXo10&position=3"
                rel="noreferrer"
                target="_blank"
              >
                See You at the Top (Flight Mix)
              </a>
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
                href="https://www.youtube.com/channel/UCPw7S3ZvjuPnc2-KCIxZYbw"
                rel="noreferrer"
                target="_blank"
              >
                Mark Sparling - Topic
              </a>
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Published on 9/28/2023, 8:18:04 PM
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 css-1j8pgxe-MuiPaper-root"
      >
        <div
          class="MuiBox-root css-p58oka"
        >
          <img
            alt="Boat Buds (Alternate Mix) thumbnail"
            class="css-wnbi1d"
            loading="lazy"
            src="https://i.ytimg.com/vi/TnsmKZmw_QY/default.jpg"
          />
          <div
            class="MuiBox-root css-gqssyb"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
                href="https://www.youtube.com/watch?v=TnsmKZmw_QY&list=OLAK5uy_lVHKpWYVeLJEP2eqMjfCk0uYE54qRXo10&position=4"
                rel="noreferrer"
                target="_blank"
              >
                Boat Buds (Alternate Mix)
              </a>
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
                href="https://www.youtube.com/channel/UCPw7S3ZvjuPnc2-KCIxZYbw"
                rel="noreferrer"
                target="_blank"
              >
                Mark Sparling - Topic
              </a>
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Published on 9/28/2023, 8:18:04 PM
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
